api/src/main/java/org/openmrs/api/db/hibernate/HibernateCohortDAO.java

```diff
@@ -59,35 +59,43 @@ public Cohort getCohort(Integer id) throws DAOException {
 	 * @see org.openmrs.api.db.CohortDAO#getCohortsContainingPatientId(java.lang.Integer)
 	 */
 	@SuppressWarnings("unchecked")
-	public List<Cohort> getCohortsContainingPatientId(Integer patientId, Boolean voided) throws DAOException {
+	public List<Cohort> getCohortsContainingPatientId(Integer patientId, Boolean includeVoidedCohorts,
+													  Boolean includeVoidedMemberships, Date asOfDate) throws DAOException {
 		Disjunction orDisjunction = Restrictions.disjunction();
 		orDisjunction.add(Restrictions.isNull("m.endDate"));
-		orDisjunction.add(Restrictions.gt("m.endDate", new Date()));
+		orDisjunction.add(Restrictions.gt("m.endDate", asOfDate));
 		Patient p = new Patient(patientId);
-		return (List<Cohort>) sessionFactory.getCurrentSession().createCriteria(Cohort.class)
-				.add(Restrictions.eq("voided", voided))
-				.createAlias("members", "m")
-				.add(Restrictions.le("m.startDate", new Date()))
-				.add(orDisjunction)
-				.add(Restrictions.eq("m.patient", p))
-				.setResultTransformer(CriteriaSpecification.DISTINCT_ROOT_ENTITY)
-				.list();
+		
+		Criteria criteria = sessionFactory.getCurrentSession().createCriteria(Cohort.class);
+		criteria.createAlias("memberships", "m");
+		criteria.add(Restrictions.not(Restrictions.gt("m.startDate", asOfDate)));
```

Comment:
Why did you change this from using `Restrictions.le("m.startDate), asOfDate)`? It's cleaner and more readable